sway/input/seatop_down.c

```diff
@@ -110,3 +129,26 @@ void seatop_begin_down(struct sway_seat *seat, struct sway_container *con,
 	container_raise_floating(con);
 	transaction_commit_dirty();
 }
+
+void seatop_begin_down_on_layer_surface(struct sway_seat *seat,
```

Comment:
There's no actual constraint as far as I can tell that this `wlr_surface` needs to be a layer surface. We can call this `seatop_begin_down_on_surface` and call it from `seatop_begin_down`, to reduce code duplication.